packages/EasyCore/src/Doctrine/Type/DateTimeMicrosecondsType.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyCore\Doctrine\Type;
+
+use DateTime;
+use DateTimeInterface;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Types\ConversionException;
+use Doctrine\DBAL\Types\Type;
+
+final class DateTimeMicrosecondsType extends Type
+{
+    private const TYPENAME = 'datetime';
+
+    /**
+     * @param mixed $value
+     *
+     * @throws \Doctrine\DBAL\Types\ConversionException
+     */
+    public function convertToDatabaseValue($value, AbstractPlatform $platform): ?string
+    {
+        if ($value === null) {
+            return $value;
+        }
+
+        if ($value instanceof DateTimeInterface) {
+            return $value->format('Y-m-d H:i:s.u');
```

Comment:
Let's move this to a `FORMAT_PHP_DATETIME` const and reuse everywhere.

Reply:
Done.